src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BookingService.kt

```diff
@@ -339,10 +339,20 @@ class BookingService(
 
       val newestSubmittedOnlineApplication = applicationService.getApplicationsForCrn(crn, ServiceName.approvedPremises)
         .filter { it.submittedAt != null }
-        .maxByOrNull { it.submittedAt!! }
+        .maxByOrNull { it.submittedAt!! } as ApprovedPremisesApplicationEntity?
       var newestOfflineApplication = applicationService.getOfflineApplicationsForCrn(crn, ServiceName.approvedPremises)
         .maxByOrNull { it.createdAt }
 
+      var newestPlacementRequest = newestSubmittedOnlineApplication?.getLatestPlacementRequest()
+
+      if (newestPlacementRequest?.isWithdrawn == true) {
+        return@validated generalError("placementRequestIsWithdrawn")
+      }
+
+      if (newestPlacementRequest?.booking != null) {
+        return@validated newestPlacementRequest.booking!!.id hasConflictError "A Booking has already been made for this Placement Request"
+      }
```

Comment:
Thinking about this, rather than returning an error, we should just not update the placement request if either of these conditionals is true. Something like (on line 406):

```kotlin
if (newestPlacementRequest != null && newestPlacementRequest?.isWithdrawn != true && newestPlacementRequest?.booking == null) {
    newestPlacementRequest.booking = booking
    placementRequestRepository.save(newestPlacementRequest)
}
```

Then update the unit tests to ensure that `mockPlacementRequestRepository.save` does not get called, instead of checking for errors.

That way the user can still create the manual booking without any unexpected behaviour or weirdness.